src/TesApi.Web/Management/Batch/TerraBatchPoolManager.cs

```diff
@@ -146,6 +144,28 @@ public async Task DeleteBatchPoolAsync(string poolId, CancellationToken cancella
             }
         }
 
+        private async Task<Guid> GetWsmResourceIdFromBatchPoolMetadataAsync(string poolId, CancellationToken cancellationToken)
+        {
+            var batchClient = CreateBatchClientFromOptions();
+
+            var pool = await batchClient.PoolOperations.GetPoolAsync(poolId, cancellationToken: cancellationToken);
+
+            if (pool is null)
+            {
+                throw new InvalidOperationException($"The batch pool was not found. Pool id:{poolId}");
+            }
+
+            var metadataItem = pool.Metadata.SingleOrDefault(m => m.Name.Equals(TerraResourceIdMetadataKey));
+
+            if (metadataItem is null || string.IsNullOrEmpty(metadataItem.Value))
```

Comment:
if (string.IsNullOrEmpty(metadataItem?.Value))

Reply:
No need to check for null on the second part of `or` the statement - short-circuit evaluation in C#

Reply:
Never mind, I miss read the suggestion. Makes sense. I agree is better.